domain/annotation/service/service.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/annotations"
+)
+
+// State describes retrieval and persistence methods for annotations.
+type State interface {
+	// GetAnnotations retrieves all the annotations associated with a given ID.
+	// If no annotations are found, an empty map is returned.
+	GetAnnotations(ctx context.Context, ID annotations.ID) (map[string]string, error)
+
+	// SetAnnotations associates key/value annotation pairs with a given ID.
+	// If annotation already exists for the given ID, then it will be updated with
```

Comment:
Just reading these comments and I am confused because of the name of the function.

`Set` implies that we will overwrite anything that is currently set with the new values. Update implies anything that isn't change will stay in place. Is it possible to update the comment and potentially the func name to reflect the side effects.